libsvmdata/tests/test_datasets.py

```diff
@@ -1,45 +1,70 @@
+import pytest
 import numpy as np
-
+from numpy.testing import assert_
 from libsvmdata import fetch_libsvm
 
-
-# TODO: test all datasets without timeout
-# @pytest.mark.parametrize("name", dataset.NAMES.keys())
-# def test_datasets(name):
-#     if "sector" in name:
-#         pytest.xfail(name)
-
-#     X, y = fetch_libsvm(name)
-#     np.testing.assert_equal(X.shape[0], y.shape[0])
+TEST_DATASETS = {
+    "regression": [
+        ("abalone", 4_177, 8),
+        ("bodyfat", 252, 14),
+    ],
+    "binary": [
+        ("a1a", 1_605, 123),
+        ("breast-cancer", 683, 10),
+    ],
+    "multiclass": [
+        ("dna", 2_000, 180),
+        ("iris", 150, 4),
+    ],
+    "multilabel": [
+        ("bibtex", 7_395, 1_836),
+        ("scene-classification", 1_211, 294),
+    ],
+}
 
 
-def test_binary():
-    # download if not present:
-    X, y = fetch_libsvm("news20.binary")
-    np.testing.assert_equal(X.shape[0], y.shape[0])
-    # also checks that loading saved files works:
-    X, y = fetch_libsvm("news20.binary")
+@pytest.mark.filterwarnings("ignore:FutureWarning")
+@pytest.mark.parametrize("dataset_name,n,p", TEST_DATASETS["regression"])
+def test_regression(dataset_name, n, p):
+    X, y = fetch_libsvm(dataset_name)
+    assert_(X.shape[0] == n)
```

Comment:
@Badr-MOUFAD assertions with np.testing assert_equal raise more explicit error messages when failing

Reply:
That's right @mathurinm!
However, we agreed on using using `assert_`

I unresolved the corresponding discussion. Refer https://github.com/mathurinm/libsvmdata/pull/37#discussion_r1108914959

Reply:
there was a misunderstanding. I did not known `assert_` existed, I was referring to `assert_*` functions such as `assert_array_less` or `assert_array_equal`

An explicit and precise feedback is essential in case of test failures, so let's use np.testing_assert_equal here

Reply:
@mathurinm @Badr-MOUFAD This should be ok now